File: lib/nodejs/benchmarks.js
const db_sync = require("./sync");
const db_async = require("./async");

host = "localhost";
port = 3434;
username = "admin";
password = "password";

dba = new db_async(host, port, username, password);

dbs = new db_sync(host, port, username, password);

var log = (d) => {console.log(d)};

//test functions
var a = (db) => {
    for (let i = 0; i < 1000; i++) {
        var path = "/" + i.toString();
        db.push(path, "RANDOM DATA", (data) => {log(data.toString())});
        db.get(path, (data) => {log(data.toString())});
        db.delete(path, (data) => {log(data.toString())}); 
    }
}

var s = (db) => {
    for (let i = 0; i < 1000; i++) {
        var path = "/" + i.toString();
        log(db.push(path, "RANDOM DATA"));
        log(db.get(path));
        log(db.delete(path));
    }
}

// console.time("Sync-db");
// s(dbs);
// console.timeEnd("Sync-db");

console.time("Async-db");
a(dba);
console.timeEnd("Async-db");

File: lib/C++/benchmarks.cc
#include <iostream>
#include "db.hpp"

using namespace std;

void single_tests(Db dba){
    for (size_t i = 0; i < 1000; i++)
    {
        std::string path = "/" + std::to_string(i);
        std::cout << dba.push(path, "\"RANDOM DATAT\"", "false") << std::endl;
        std::cout << dba.get(path) << std::endl;
        std::cout << dba.remove(path) << std::endl;
    }
}


int main(){
    Db database("localhost", 3434, "admin", "password");
    single_tests(database);
    return 0;
}

/* 1000 write/read/delete with time ./bench
real    0m2.861s
user    0m0.062s
sys     0m0.101s
*/

/* 10000 write/read/delete with time ./bencht
real    0m2.732s
user    0m0.060s
sys     0m0.099s
*/